### PR TITLE
Migrate to py3

### PR DESCRIPTION
**Pull request by razzeee**
This is untested as of now.

We need to run https://gitlab.com/ronie/script.language-convert/-/tree/master and https://gitlab.com/ronie/script.settings-convert against this, to migrate those. But I don't have a matrix setup on my linux box at the moment.

### CODE REVIEW

**Comment by razzeee** [comment]
Should I also setup a addon submission workflow?

**Comment by bossanova808** [comment]
Err, well this addon checker thing (and indeed the ronie scripts) are new to me - previous things I have done by hand, and I have a bit of a recipe for the addon submissions but if there is a proper/more modern way to do it with workflows, I'm happy to learn!

**Comment by bossanova808** [comment]
(You're more than welcome to take this over if you want, BTW - I'm happy to keep it working via PRs and the like as time does by but my total available time for Kodi is not nearly what I'd like so that might make more sense...but also happy to learn and carry on if you can deal with me being slow...)

**Comment by razzeee** [comment]
I don't think I want to take it over :)

I can add the submission tooling in another PR after we tested and merged this.
https://github.com/xbmc/action-kodi-addon-submitter

File: addon.xml [hunk 1]
```diff
@@ -1,11 +1,11 @@
 <?xml version="1.0" encoding="UTF-8" standalone="yes"?>
-<addon id="script.service.checkpreviousepisode" name="XBMC Check Previous Episode" version="0.3.5" provider-name="Lucleonhart, bossanova808">
+<addon id="script.service.checkpreviousepisode" name="XBMC Check Previous Episode" version="0.4.0" provider-name="Lucleonhart, bossanova808">
```
Comment (razzeee) [inline addon.xml]: I bumped to the next minor version

File: default.py [hunk 2]
```diff
@@ -160,77 +179,67 @@ def manageIgnored():
...
     player_monitor = MyPlayer()
-    while not xbmc.abortRequested:
-          xbmc.sleep(100)
-
-
-
-
-
 
+    while not monitor.abortRequested():
```
Comment (razzeee) [inline default.py]: This is the new handling for abortRequested

File: default.py [hunk 3]
```diff
@@ -160,77 +179,67 @@ def manageIgnored():
...
 else:
-
-    log( "Version: %s Started" % (__version__), xbmc.LOGNOTICE)
-    # # Kodi Krypton and below:
-    if __kodiversion__ < 17.9:
-        log('Kodi ' + str(__kodiversion__) + ', listen to onPlayBackStarted', xbmc.LOGNOTICE)
-    # Kodi Leia and above:
-    else:
-        log('Kodi ' + str(__kodiversion__) + ', listen to onAVStarted', xbmc.LOGNOTICE)
+    log('Kodi ' + str(__kodiversion__) +
```
Comment (razzeee) [inline default.py]: I took the liberty to remove the handling for versions earlier then leia

File: default.py [hunk 4]
```diff
@@ -160,77 +179,67 @@ def manageIgnored():
...
 
 # Listen to appropriate events for different Kodi versions
-class MyPlayer( xbmc.Player ):
-
-    def __init__( self, *args, **kwargs ):
-        xbmc.Player.__init__( self )
+class MyPlayer(xbmc.Player):
 
-    def onPlayBackStarted( self ):
-        if __kodiversion__ < 17.9:
```
Comment (razzeee) [inline default.py:198]: Removed this as kodi 19 will never trigger this

**Comment by bossanova808** [comment]
Yep all the changes look fine really.

I'll do some testing during the week (I don't have matrix working here yet...but I do on my work setup IIRC (it's ok - I own the company ;) )

This new action submission thing seems a great way of doing it vs. the git malarky which is a bit involved and easy to go wrong with...so yep, let's definitely do that once I've merged all this.

(I should get Matrix going here too really.... but thanks to this bloody pandemic I'm having to juggle way more stuff than usual :( )

**Comment by razzeee** [comment]
Should be good now. I even tested it quickly.

**Comment by bossanova808** [comment]
Ok, great job, all seems to be working perfectly.

Did you want to add the workflow for addon submission now then?

Also - you should pop your name in the contributors.  And normally I'd update the `changelog.txt` but I think there was talk of moving away from that?  Not sure what the outcome was there.

**Comment by razzeee** [comment]
I will send a PR with the submission flow soon.

Yes, changelog is not mandatory, I still maintain them for my projects.

**Comment by bossanova808** [comment]
Fantastic, thanks.  I fixed some minors with the addon.xml changes (thanks to your workflow) and updated the changelog & contributors.  So basically with a submissions workflow, and then a tag, we're all good to go I think.

This will be super helpful with my other addons, too - very much appreciated!

**Pull request merged**